File: backend/src/Hrms.Domain/Entities/Shift.cs
using System;

namespace Hrms.Domain.Entities;

public class Shift
{
    public long Id { get; set; }
    public string ShiftCode { get; set; } = string.Empty;
    public string ShiftName { get; set; } = string.Empty;
    public TimeOnly StartTime { get; set; }
    public TimeOnly EndTime { get; set; }
    public bool IsCrossDay { get; set; } = false;
    public int BreakMinutes { get; set; } = 0;
    public string Status { get; set; } = "ACTIVE"; // 'ACTIVE', 'INACTIVE'
    public int LateGraceMinutes { get; set; } = 0;
    public int EarlyLeaveGraceMinutes { get; set; } = 0;
}

File: frontend/src/types/shift.ts
export interface Shift {
  id: number;
  shiftCode: string;
  shiftName: string;
  startTime: string; // HH:mm
  endTime: string;   // HH:mm
  isCrossDay: boolean;
  breakMinutes: number;
  status: 'ACTIVE' | 'INACTIVE';
  lateGraceMinutes: number;
  earlyLeaveGraceMinutes: number;
  workHours: number;
  netWorkHours: number;
}

export interface CreateShiftRequest {
  shiftCode: string;
  shiftName: string;
  startTime: string; // HH:mm
  endTime: string;   // HH:mm
  isCrossDay?: boolean;
  breakMinutes: number;
  status: string;
  lateGraceMinutes: number;
  earlyLeaveGraceMinutes: number;
}

export interface UpdateShiftRequest {
  shiftName: string;
  startTime: string;
  endTime: string;
  isCrossDay?: boolean;
  breakMinutes: number;
  status: string;
  lateGraceMinutes: number;
  earlyLeaveGraceMinutes: number;
}
